activestorage/lib/active_storage/attached/changes/create_one.rb

```diff
@@ -59,13 +59,15 @@ def find_or_build_blob
             filename: attachable.original_filename,
```

Comment:
In above lines we define that an attachment built from another attachment does not create a new blob:

```ruby
      def find_or_build_blob
        case attachable	        case attachable
        when ActiveStorage::Blob	        when ActiveStorage::Blob
          attachable	          attachable
        when ActionDispatch::Http::UploadedFile, Rack::Test::UploadedFile	        when ActionDispatch::Http::UploadedFile, Rack::Test::UploadedFile
          ActiveStorage::Blob.build_after_unfurling(	          ActiveStorage::Blob.build_after_unfurling(
            io: attachable.open,	            io: attachable.open,
            filename: attachable.original_filename,	            filename: attachable.original_filename,
```

That means that the following situation is possible:

```
# user1.photo.attach(user2.photo_blob)
# user1.photo_blob == user2.photo_blob
=> true
# user1.photo.attach(io: …)

Now, user2 has a blob with a key generated for user1, which has no association to user1. This becomes worse if the objects belong to different models. How do we plan to tackle that?

Reply:
In this context it comes down to a business decision.

In most cases, the filename isn't relevant. So it doesn't matter.

In the cases were it *is* relevant, I can't think of any instance where we copy an existing blob. We're just creating new ones. The primary use-case are generated project documents and call recordings. We always generate NEW docs in these cases.

Reply:
Thanks @xjunior,

I think as @wadewinningham said, it comes down to a business decision but here are some thought about your concern!

- The previous current behaviour has the same issue, meaning both users have a Blob with the same unique key (which btw isn't unique anymore).

- I guess you could do what you need [like that](https://edgeapi.rubyonrails.org/classes/ActiveStorage/Blob.html#method-i-open):
```ruby
user2.photo_blob.open do |file|
  user1.photo.attach(io: file) # => that will generate the right key or you can specify one yourself!
end
```